tools/swift-reflection-fuzzer/swift-reflection-fuzzer.cpp

```diff
@@ -73,6 +74,15 @@ class ObjectMemoryReader : public MemoryReader {
       *result = sizeof(void *);
       return true;
     }
+    case DLQ_GetPtrAuthMask: {
+      auto result = static_cast<uintptr_t *>(outBuffer);
+#if __has_feature(ptrauth_calls)
+      *result = (uintptr_t)ptrauth_strip((void*)0x0007ffffffffffff, 0);
```

Comment:
Do we have a true mask that we can rely on instead of hard coding?

Reply:
Can you add a comment here explaining what you are doing in more detail. That is can you explain here what you are actually testing.

Reply:
Also you should run this through git-clang-format. The reason why I know that you haven't is that `(void*)` should be `(void *)`. Also can you use static_cast like the rest of the code rather than using c casts?

Reply:
Also I think that __has_feature is a clang extension.

Reply:
All of the above is true, however I copied this piece of code directly from `include/swift/Remote/InProcessMemoryReader.h` (apparently the original was written by @kubamracek).

Not sure where that leaves us exactly. Should I fix that file at the same time?

Reply:
As far as `__has_feature()` goes, it looks like we use it all over the shop, and various files have their own variant of

    #ifndef __has_feature
    #define __has_feature(x) 0
    #endif

which arguably belongs in `swift/Basic/Compiler.h` with the other similar defines.